### DISCUSSION TITLE
[Feature] Set user's storage label based on OIDC claim

### DISCUSSION THREAD

**Discussion by pixil98** [immich-app/immich#3268] [Feature Request]
### The feature

I recently setup Immich using Authentik as the identity provider. I would like the ability to use a claim to automatically set a storage label for the user during automatic account creation.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by etnoy** [comment]
This is an excellent suggestion

**Comment by jrasm91** [comment]
How are you imagining this working? A setting in Administration > Settings > OAuth? Something like a text field for "Storage Label Attribute"? If it is set, the value of the claim would be automatically synced with `user.storageLabel`?

**Reply by pixil98** [reply]
Yes, that's exactly what I'd envision. I already know that the usernames in authentik are unique so i'd like to set them automatically as the storage label. I should be able to pull them from the preferred_username claim.  I could also see it being helpful to be able to specify which scopes Immich requests in case you want to control the claims on the token via a separate scope (I don't need this for my setup).

**Reply by jrasm91** [reply]
You can already specify custom scopes

**Comment by pixil98** [comment]
You sure can. That what I get for responding at work without access to my deployment.

**Comment by jrasm91** [comment]
Converted to an issue (#3276) to track progress. Being implemented in PR #3278.
